src/control/jobs/control_jobs.c

```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
                                                           FALSE));
 }
 
+static gboolean _same_basefilename(const char *filename, const char *prev_filename)
```

Comment:
I would put that in utility.h and renamed to : `dt_has_same_base_filename()`

And I will make use of `g_path_get_basename)` to simplify the implementation.

Reply:
> I would put that in utility.h and renamed to : dt_has_same_base_filename()

I'll do it.

> And I will make use of g_path_get_basename) to simplify the implementation.

g_path_get_basename() returns the last component of the filename. I don't see how I can use this here.

Reply:
I see, this is not checking **basename** but **full pathname** without extension. The routine name is confusing then.

Reply:
I see that you have renamed it to: `dt_has_same_path_basename()`, so this convey better the implementation.